src/router_core/transfer.c

```diff
@@ -634,6 +634,23 @@ static void qdr_link_forward_CT(qdr_core_t *core, qdr_link_t *link, qdr_delivery
             addr->deliveries_ingress++;
         link->total_deliveries++;
     }
+    //
+    // There is no address that we can send this delivery to, which means the addr was not found in our hastable. This
+    // can be because there were no receivers or because the address was not defined in the config file.
+    // If the treatment for such addresses is set to be forbidden, we send back a rejected disposition and detach the link
+    //
+    else if (core->qd->treatment == QD_TREATMENT_LINK_FORBIDDEN) {
+        dlv->disposition = PN_REJECTED;
+        dlv->error = qdr_error("qd:forbidden", "Sending deliveries to this address is forbidden");
```

Comment:
You should be using one of the AMQP-specified error conditions that is in the amqp.[ch] files.  It seems that the appropriate error condition is "amqp:not-allowed", not "forbidden" (which is an error code only for the management protocol).